src/device/pf_cmrpc.c

```diff
@@ -2473,45 +2479,54 @@ static int pf_cmrpc_dce_packet(
          }
 
          /* Enter here _even_if_ an error is already detected because we need to generate an error response. */
-         if (p_sess->fragment_nbr != rpc_req.fragment_nmb)
+         if (p_sess->in_fragment_nbr != rpc_req.fragment_nmb)
          {
             LOG_ERROR(PF_RPC_LOG, "CMRPC(%d): Unexpected fragment number %u != %u (exp)\n",
-               __LINE__, (unsigned)rpc_req.fragment_nmb, (unsigned)p_sess->fragment_nbr + 1);
+               __LINE__, (unsigned)rpc_req.fragment_nmb, (unsigned)p_sess->in_fragment_nbr + 1);
             pf_set_error(&p_sess->rpc_result, PNET_ERROR_CODE_CONNECT, PNET_ERROR_DECODE_PNIO, PNET_ERROR_CODE_1_CMRPC, PNET_ERROR_CODE_2_CMRPC_STATE_CONFLICT);
          }
-         else  if ((p_sess->buf_len + rpc_req.length_of_body) > sizeof(p_sess->buffer))
+         else  if ((p_sess->in_buf_len + rpc_req.length_of_body) > sizeof(p_sess->in_buffer))
          {
             LOG_ERROR(PF_RPC_LOG, "CMRPC(%d): Fragments exceed max buffer\n", __LINE__);
             pf_set_error(&p_sess->rpc_result, PNET_ERROR_CODE_CONNECT, PNET_ERROR_DECODE_PNIO, PNET_ERROR_CODE_1_CMRPC, PNET_ERROR_CODE_2_CMRPC_STATE_CONFLICT);
          }
          else
          {
             /* Copy to session buffer */
-            memcpy(&p_sess->buffer[p_sess->buf_len], &p_req[req_pos], rpc_req.length_of_body);
-            p_sess->buf_len += rpc_req.length_of_body;
-            p_sess->fragment_nbr++;
+            memcpy(&p_sess->in_buffer[p_sess->in_buf_len], &p_req[req_pos], rpc_req.length_of_body);
+            p_sess->in_buf_len += rpc_req.length_of_body;
+            p_sess->in_fragment_nbr++;
 
             if (rpc_req.flags.last_fragment == true)
             {
                /* Re-route the parser to use the session buffer */
                req_pos = 0;
-               p_sess->get_info.p_buf = p_sess->buffer;
-               p_sess->get_info.len = p_sess->buf_len;
+               p_sess->get_info.p_buf = p_sess->in_buffer;
+               p_sess->get_info.len = p_sess->in_buf_len;
             }
          }
       }
 
-      /* Enter here _even_if_ an error is already detected because may we need to generate an error response. */
+      /* Enter here _even_if_ an error is already detected because we may need to generate an error response. */
       if ((rpc_req.flags.fragment == false) ||
           (rpc_req.flags.last_fragment == true))
       {
 
          switch (rpc_req.packet_type)
          {
          case PF_RPC_PT_REQUEST:
+            /* A new request - clear the response buffer */
+            p_sess->out_buf_len = 0;
+            p_sess->out_buf_sent_len = 0;
+            p_sess->out_fragment_nbr = 0;
+
             pf_get_ndr_data(&p_sess->get_info, &req_pos, &p_sess->ndr_data);
             p_sess->get_info.is_big_endian = true;  /* From now on all is big-endian */
 
+            if (max_rsp_len > p_sess->ndr_data.args_maximum)
```

Comment:
In my application this limitation results in a demaged response for the ParameterEnd request.
The payload of the response used here is 32 bytes in this case (I use a S7-1212).
So the complete response frame is limited to 32bytes as well. But it would need 174 bytes, just the payload is 32 bytes.
It seems that the payload and the framesize is mixe here.

With this limitation commented ut all works fined...